ui/src/components/WorkerDetailsCard/index.jsx

```diff
@@ -1,52 +1,189 @@
+      });
+    } catch (error) {
+      this.handleDialogActionError(error);
+    }
+  };
+
+  handleDialogActionError = error => {
+    this.setState({
+      dialogState: {
+        ...this.state.dialogState,
+        error,
+      },
+    });
+  };
+
+  handleDialogActionClose = () => {
+    this.setState({
+      dialogState: {
+        ...this.state.dialogState,
+        error: null,
+        open: false,
+      },
+    });
+  };
+
   render() {
     const {
-      worker: { quarantineUntil, firstClaim, lastDateActive },
+      worker: {
+        quarantineUntil,
+        firstClaim,
+        lastDateActive,
+        providerId,
+        taskQueueId,
+        workerGroup,
+        workerId,
+      },
     } = this.props;
+    const {
+      dialogState: { open, error, title, confirmText, body },
+    } = this.state;
+    const iconSize = 16;
 
     return (
-      <List>
-        <ListItem>
-          <ListItemText
-            primary="Last Active"
-            secondary={
-              lastDateActive ? <DateDistance from={lastDateActive} /> : 'n/a'
-            }
-          />
-        </ListItem>
-        <ListItem>
-          <ListItemText
-            primary="First Claim"
-            secondary={<DateDistance from={firstClaim} />}
-          />
-        </ListItem>
-        <ListItem>
-          <ListItemText
-            primary="Quarantine Until"
-            secondary={
-              quarantineUntil
-                ? format(parseISO(quarantineUntil), 'yyyy/MM/dd')
-                : 'n/a'
-            }
-          />
-        </ListItem>
-      </List>
+      <Fragment>
+        <List>
+          <ListItem>
+            <ListItemText
+              primary="Last Active"
+              secondary={
+                lastDateActive ? <DateDistance from={lastDateActive} /> : 'n/a'
+              }
+            />
+          </ListItem>
+          <ListItem>
+            <ListItemText
+              primary="First Claim"
+              secondary={<DateDistance from={firstClaim} />}
+            />
+          </ListItem>
+          <ListItem>
+            <ListItemText
+              primary="Quarantine Until"
+              secondary={
+                quarantineUntil
+                  ? format(parseISO(quarantineUntil), 'yyyy/MM/dd')
+                  : 'n/a'
+              }
+            />
+          </ListItem>
+          {providerId !== NULL_PROVIDER ? (
+            <ListItem>
+              <Button
+                variant="outlined"
+                endIcon={<DeleteIcon size={iconSize} />}
+                onClick={this.handleDialogActionOpen(
+                  taskQueueId,
+                  workerGroup,
+                  workerId
+                )}>
+                Terminate Worker
+              </Button>
+            </ListItem>
+          ) : (
+            <ListItem>
+              <Label mini status="warning">
+                Scheduled for termination
+              </Label>
+            </ListItem>
+          )}
```

Comment:
this came out of nowhere :) 
if provider is not null provider -> display "Delete" button, else -> worker is terminated? Are we jumping into conclusions about `worker.state` here?  :)

Reply:
Speaking of states.. I don't think we show here worker state at all, should we add this?

Reply:
This is one of the things I'd love to work alongside you with to make sure we get it right! :)